pandas/tests/groupby/transform/test_transform.py

```diff
@@ -1289,3 +1289,13 @@ def test_transform_cumcount():
 
     result = grp.transform("cumcount")
     tm.assert_series_equal(result, expected)
+
+
+def test_group_mean_timedelta_nat():
+    series = Series(["1 day", "3 days", pd.NaT], dtype="timedelta64[ns]")
+
+    group = series.groupby([1, 1, 1])
+    result = group.transform("mean")
```

Comment:
mean is an aggregation, not a transformation, so this would go in tests/groupby/aggregate/...

Reply:
Thanks for the advice, in hindsight makes total sense, I put it here because the original linked bug, #43132, had used a transform keyword but I see that that's not what we mean here in the test structure